agent/fw_drupal.c

```diff
@@ -717,6 +732,35 @@ NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke) {
 }
 NR_PHP_WRAPPER_END
 
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
+     && !defined OVERWRITE_ZEND_EXECUTE_DATA
+NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke_all_before) {
+  (void)wraprec;
+  zval* hook_copy = NULL;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_DRUPAL);
+
+  hook_copy = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS);
+  if (nr_php_is_zval_non_empty_string(hook_copy)) {
+    nr_stack_push(&NRPRG(drupal_module_invoke_all_hooks),
+                  hook_copy);
+  }
+}
+NR_PHP_WRAPPER_END
+
+NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke_all_after) {
+  (void)wraprec;
+  /* using nr_php_get_user_func_arg() so that we don't perform another copy
+   * when all we want to do is check the string length */
+  zval* orig = nr_php_get_user_func_arg((zend_uint)1, NR_EXECUTE_ORIG_ARGS);
```

Comment:
This function is pointed to by the clean callback, but when cleaning up, we don't have access to the zend_execute_data.  We can only clean up NR globals.

Reply:
Gotcha, that makes a lot of sense. Unfortunately, that means that we are back to needing another global stack to determine whether or not we have pushed to this stack. Because an empty string/null will not be pushed to this global stack and the execute_data was previously how I was determining this.